core/job/job.go

```diff
@@ -34,14 +34,58 @@ func (j Job) Spec() *Spec {
 	return j.spec
 }
 
-func (j Job) GetName() string { // to support multiroot DataTree
+func (j Job) GetName() string {
 	return j.spec.name.String()
 }
 
 func (j Job) FullName() string {
 	return j.ProjectName().String() + "/" + j.spec.name.String()
 }
 
+func (j Job) GetJobWithUnresolvedUpstream() (*WithUpstream, error) {
```

Comment:
let's add test cases for `GetJobWithUnresolvedUpstream` and `GetJobsWithUnresolvedUpstreams`